src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,101 @@
-export const Pagination = () => {};
+import React, { useEffect } from 'react';
+import { getPartItems } from '../../utils';
+
+type Props = {
+  items: string[];
+  total: number;
+  perPage: number;
+  currentPage: number;
+  setElements: React.Dispatch<React.SetStateAction<string[]>>;
+  onPageChange: React.Dispatch<React.SetStateAction<number>>;
+};
+
+export const Pagination: React.FC<Props> = ({
+  items,
+  total,
+  perPage,
+  currentPage,
+  setElements,
+  onPageChange,
+}) => {
+  const amountOfButton = Math.ceil(total / Number(perPage));
+  const buttons = [];
+
+  const handlePageClick = (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+    pageNumber: number,
+  ) => {
+    event.preventDefault();
+    onPageChange(pageNumber);
+  };
+
+  useEffect(() => {
+    const updatedElements = getPartItems(items, currentPage, perPage);
+
+    setElements(updatedElements);
+  }, [currentPage, perPage, items, setElements]);
+
+  const currentAmountOfItems = getPartItems(items, currentPage, perPage);
```

Comment:
The calculation of 'currentAmountOfItems' is redundant since the same calculation is performed inside the useEffect hook on line 32. You can remove this line and use the elements from the state directly within your render logic.